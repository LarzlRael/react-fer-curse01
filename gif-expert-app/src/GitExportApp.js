import React, { useState } from 'react'
import AddCategory from './AddCategory';
import GifGrid from './components/GifGrid';

const GitExportApp = ({ defaultCategories = [] }) => {

    //const categories = ['Dragon Ball', 'naruto', 'my hero academia']
    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     // add a new category by spreed operator
    //     // 1 way
    //     //? setCategories([...categories, 'one piece']);
    //     //2 way
    //     setCategories(cats => [...cats, 'one piece']);

    // }
    return (
        <>
            <h2>GiExportapp</h2>
            <AddCategory
                setCategories={setCategories}
            />

            <hr />


            <ol>
                {categories.map(category => (
                    <GifGrid category={category} key={category} />
                ))}
            </ol>
        </>
    )
}

export default GitExportApp
