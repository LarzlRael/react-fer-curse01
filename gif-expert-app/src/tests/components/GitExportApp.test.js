import React from 'react';
import { shallow } from "enzyme"
import GitExportApp from "../../GitExportApp";


describe('Testing the git Expert App Component', () => {

    const wrapper = shallow(<GitExportApp />)
    test('should show correctly the component with props', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('should to show a category list items', () => {

        const categories = ['One Punch', 'Dragon Ball'];

        const wrapper = shallow(<GitExportApp defaultCategories={categories} />)

        // Expect gifGrid return the same length
        expect(wrapper.find('GifGrid').length).toBe(categories.length);



    });
})
