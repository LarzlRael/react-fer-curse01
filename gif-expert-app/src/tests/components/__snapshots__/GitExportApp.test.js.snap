// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the git Expert App Component should show correctly the component with props 1`] = `
<Fragment>
  <h2>
    GiExportapp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
